pkg/resources/deviceSelectors.go

```diff
@@ -64,6 +65,28 @@ func (s *driverSelector) Filter(inDevices []types.PciNetDevice) []types.PciNetDe
 	return filteredList
 }
 
+// newNetDevSelector returns a NetDevSelector interface for netDev list
+func newNetDevSelector(netDevs []string) types.DeviceSelector {
+	return &netDevSelector{netDevs: netDevs}
+}
+
+type netDevSelector struct {
+	netDevs []string
+}
+
+func (s *netDevSelector) Filter(inDevices []types.PciNetDevice) []types.PciNetDevice {
+	filteredList := make([]types.PciNetDevice, 0)
+	for _, dev := range inDevices {
+		if netDevs, err := utils.GetNetNames(dev.GetPciAddr()); err == nil {
```

Comment:
Is it worth adding the ability to filter user space device name here, for example: dev/vfio/<num>?

Reply:
For Mellanox card, this may not be necessary as there is no pre binding of devices to vfio or uio driver. @ahalim-intel do you think we need a user space device name filter for other NICs?

Reply:
With userspace driver there won't be any kernel interface name available.

Reply:
Yes, there won't be kernel interface name for userspace driver, what about the device spec name ``/dev/vfio/<num>``? is it worth filtering? 
Another problem(also discussed in #109 ) is that the ``netDevs`` filter may contain a node specific interface name that may not applied to other  nodes in a heterogeneous environment.